### PR TITLE
PWN-4268 Phone Number fixes

### PR DESCRIPTION
**Pull request by kuznetsov**
### Jira Tickets

https://p2pvalidator.atlassian.net/browse/PWN-4268
https://p2pvalidator.atlassian.net/browse/PWN-4485
https://p2pvalidator.atlassian.net/browse/PWN-4486
https://p2pvalidator.atlassian.net/browse/PWN-4487
https://p2pvalidator.atlassian.net/browse/PWN-4490
https://p2pvalidator.atlassian.net/browse/PWN-4491
https://p2pvalidator.atlassian.net/browse/PWN-4492
https://p2pvalidator.atlassian.net/browse/PWN-4493
https://p2pvalidator.atlassian.net/browse/PWN-4495

### Description of Work

Design fixes on `PhoneNumberEnterFragment` and `CountryCodePickerDialog`

### Screenshots

Screenshot 2022-08-18 at 17 30 29

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/auth/ui/phone/countrypicker/CountryCodePickerPresenter.kt [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 
 import kotlinx.coroutines.launch
-import org.p2p.wallet.auth.ui.phone.CountryCodeInteractor
 import org.p2p.wallet.auth.model.CountryCode
 import org.p2p.wallet.auth.model.CountryCodeItem
+import org.p2p.wallet.auth.ui.phone.CountryCodeInteractor
 import org.p2p.wallet.common.mvp.BasePresenter
 import org.p2p.wallet.utils.emptyString
 
 private const val DEFAULT_KEY = ""
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/auth/ui/phone/countrypicker/CountryCodePickerPresenter.kt:10]: private const val DEFAULT_KEY = emptyString()

Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/auth/ui/phone/countrypicker/CountryCodePickerPresenter.kt:10]: `private const val` can't be a function result

File: app/src/main/java/org/p2p/wallet/auth/ui/phone/countrypicker/CountryCodePickerPresenter.kt [hunk 2]
```diff
@@ -16,28 +17,52 @@ class CountryCodePickerPresenter(
...
     }
 
     override fun load(countryCode: CountryCode?) {
         launch {
-            allCountryCodeItems = countryCodeInteractor.getCountries().map { CountryCodeItem(it, false) }
+            allCountryCodeItems = countryCodeInteractor.getCountries().map {
+                CountryCodeItem(it, isSelected = it.nameCode == countryCode?.nameCode)
+            }.sortedBy {
+                !it.isSelected
+            }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/phone/countrypicker/CountryCodePickerPresenter.kt:65]:
```suggestion
            allCountryCodeItems = countryCodeInteractor.getCountries()
                .map { CountryCodeItem(it, isSelected = it.nameCode == countryCode?.nameCode) }
                .sortedBy { !it.isSelected }
```

File: app/src/main/java/org/p2p/wallet/auth/widget/AnimatedSearchView.kt [hunk 3]
```diff
@@ -22,11 +25,17 @@ class AnimatedSearchView @JvmOverloads constructor(
     private var stateListener: SearchStateListener? = null
 
     init {
-        binding.showButtonContainer.setOnClickListener { openSearch() }
-        binding.closeButton.setOnClickListener { closeSearch() }
+        binding.apply {
+            showButtonContainer.setOnClickListener { openSearch() }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/widget/AnimatedSearchView.kt:29]: can you rename views by the way?

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt [hunk 4]
```diff
@@ -58,7 +58,8 @@ class NewOnboardingFragment :
             viewPagerOnboardingSlider.adapter = BaseFragmentAdapter(childFragmentManager, lifecycle, fragments, args)
             dotsIndicatorOnboardingSlider.attachTo(viewPagerOnboardingSlider)
             buttonCreateWalletOnboarding.setOnClickListener {
-                presenter.onSignUpButtonClicked()
+                // TODO: Replace back to presenter.onSignUpButtonClicked() after QA
+                replaceFragment(PhoneNumberEnterFragment.create())
```
[the remark this answers is not in the document]
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt:62]: it is better to place ticket number for TODO

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/widget/AnimatedSearchView.kt [hunk 5]
```diff
@@ -22,11 +25,17 @@ class AnimatedSearchView @JvmOverloads constructor(
...
+            imageViewErase.setOnClickListener { editText.text.clear() }
+            closeButton.setOnClickListener { closeSearch() }
+            editText.doOnTextChanged { text, _, _, _ ->
+                imageViewErase.isVisible = text?.isNotEmpty() ?: false
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/auth/widget/AnimatedSearchView.kt:33]:
```suggestion
                imageViewErase.isVisible = !text.isNullOrEmpty()
```

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/auth/ui/phone/countrypicker/CountryCodePickerPresenter.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/auth/ui/phone/countrypicker/CountryCodePickerPresenter.kt:10]: okey but sad =(

File: app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt:62]: No need to create a ticket for that

[the remark this answers is not in the document]
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt:62]: Why do we hardcode the navigation for QA? 
They have a local docker working, do they?

[the remark this answers is not in the document]
Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt:62]: It easier to test changes in scope of single screen hardcoding it's navigation. But I'd talk to them

[the remark this answers is not in the document]
Comment (kuznetsov) [reply app/src/main/java/org/p2p/wallet/auth/ui/onboarding/NewOnboardingFragment.kt]: We decide to test all flow

**Pull request merged**